mixer/pkg/protobuf/yaml/dynamic/auth.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamic
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+
+	policypb "istio.io/api/policy/v1beta1"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	//
+	publicCACertsFile = "/etc/ssl/certs/ca-certificates.crt"
+)
+
+type authHelper struct {
+	authCfg          *policypb.Authentication
+	skipVerification bool
+}
+
+type headerToken struct {
+	header string
+	prefix string
+	token  []byte
+}
+
+func (c *headerToken) GetRequestMetadata(context.Context, ...string) (map[string]string, error) {
+	if c.prefix == "" {
+		return map[string]string{
+			c.header: string(c.token),
+		}, nil
+	}
+	return map[string]string{
+		c.header: c.prefix + " " + string(c.token),
+	}, nil
+}
+
+func (c *headerToken) RequireTransportSecurity() bool {
+	return true
```

Comment:
This line is not covered by tests. Is this needed?
https://320774-74175805-gh.circle-artifacts.com/0/go/out/codecov/pr/coverage.html#file248

Reply:
It is not executed, but it is required by oauth TokenSource interface.